app/assets/javascripts/discourse/app/controllers/flag.js

```diff
@@ -276,7 +276,9 @@ export default Controller.extend(ModalFunctionality, {
         .act(this.model, params)
         .then(() => {
           if (!params.skipClose) {
-            this.send("closeModal");
+            if (!this.isDestroying) {
```

Comment:
It's good to also check `isDestroyed`. And the `this.set()` below can also fail if the controller is gone, so this check should be moved higher:

```js
.then(() => {
  if (this.isDestroying || this.isDestroyed) {
    return;
  }

  …
```

Reply:
@CvX I've fixed it, thank you! I also added the same check to the catch block below.

By the way, looks like _technically_ there is no need to check `isDestoroyed` after checking `isDestroying`. After an object is destroyed, `isDestroying` is still true.

It's said explicitly in the [docs for the latest Ember version](https://api.emberjs.com/ember/release/functions/@ember%2Fdestroyable/isDestroying). It _isn't_ said explicitly in the [docs for our Ember version](https://api.emberjs.com/ember/3.12/classes/CoreObject/properties/isDestroying?anchor=isDestroying) but from looking at [this method](https://github.com/emberjs/ember.js/blob/cf921f10d5db4d9c179067073af4ee574034cc68/packages/%40ember/-internals/runtime/lib/system/core_object.js#L589-L595) and the code around of it looks like it it's true for our version of Ember too.

Despite this, I think it's still a good idea to be checking for `isDestroyed` explicitly in our code. This way, readers won't need to think if we should check for `isDestroyed` too and look at docs and source.